strawberryfields/apps/similarity.py

```diff
@@ -306,21 +335,133 @@ def event_cardinality(photon_number: int, max_count_per_mode: int, modes: int) -
     return cardinality
 
 
+def _get_state(graph: nx.Graph, n_mean: float = 5, loss: float = 0.0) -> BaseGaussianState:
+    r"""Embeds the input graph into a GBS device and returns the corresponding Gaussian state.
+    """
+    modes = graph.order()
+    A = nx.to_numpy_array(graph)
+    mean_photon_per_mode = n_mean / float(modes)
+
+    p = sf.Program(modes)
+
+    # pylint: disable=expression-not-assigned
+    with p.context as q:
+        sf.ops.GraphEmbed(A, mean_photon_per_mode=mean_photon_per_mode) | q
+
+        if loss:
+            for _q in q:
+                sf.ops.LossChannel(1 - loss) | _q
+
+    eng = sf.LocalEngine(backend="gaussian")
+    return eng.run(p).state
+
+
+def prob_orbit_exact(graph: nx.Graph, orbit: list, n_mean: float = 5, loss: float = 0.0) -> float:
+    r"""Gives the exact probability of a given orbit for the input graph.
+
+    The exact probability of an orbit is the sum of probabilities of
+    all possible GBS output patterns that belong to it:
+
+    .. math::
+       p(O) = \sum_{S \in O} p(S) = \frac{{\rm haf}(A_{S})^2}{S! \sqrt{\det(Q)}}
+
+    where :math:`S` represents one GBS output pattern, :math:`A_{S}` is its induced
+    sub-matrix and :math:`Q` is the :math:`Q` covariance matrix obtained from the adjacency matrix.

```

Comment:
```suggestion
       p(O) = \sum_{S \in O} p(S)

    where :math:`S` are samples belonging to :math:`O`.
```

Reply:
Just to emphasize that the last equality is wrong in its current form, it would need the sum over S in O to be correct and also ideally a clarification that S! means the product of factorials of the elements.

^ Hence I think we should keep things simple and just delete.